python/GangaLHCb/__init__.py

```diff
@@ -82,16 +83,30 @@ def _store_root_version():
 
 def _store_dirac_environment():
     from GangaDirac.Lib.Utilities.DiracUtilities import write_env_cache, get_env
-    diracversion = _guess_version('LHCBDIRAC')
     platform = os.environ['CMTOPT']
+    diracversion = Ganga.Utility.Config.getConfig('LHCb')['diracVersion']
     fdir = os.path.join(os.path.expanduser("~/.cache/Ganga/GangaLHCb"), platform)
-    fname = os.path.join(fdir, diracversion)
+    fname = os.path.join(fdir, defaultLHCbDirac)
```

Comment:
The dirac version can be specified when starting ganga - if you give it a version that doesn't exist you need to fall back on the default. This part checks that the default environment has already been saved in case there is an empty cache and you are trying to use a non-existing version of LHCbDirac.